### PR TITLE
feat: support image upload from local file system

### PR DESCRIPTION
**Pull request by thorseraq**
This PR:
- added image support for blockHub
- exposed `async uploadImageFromLocal(editor: EditorContainer)` util for image uploading from local file system 

close https://github.com/toeverything/blocksuite/issues/1186.

### CODE REVIEW

File: packages/blocks/src/components/blockhub.ts [hunk 1]
```diff
@@ -43,7 +45,7 @@ type BlockHubItem = {
   toolTip: string;
 };
 
-export type CardListType = 'blank' | 'list' | 'text' | 'database';
+export type CardListType = 'blank' | 'list' | 'text' | 'database' | 'file';
```
Comment (lawvs) [inline packages/blocks/src/components/blockhub.ts:48]: Useless update

**Review by lawvs — COMMENTED** [review]
I don't think the motivation behind adding `addBlocksByFlavour` and `addSiblingBlocks` is significant, so I don't believe they need to be included.

File: packages/store/src/workspace/page.ts [hunk 2]
```diff
@@ -461,28 +492,44 @@ export class Page extends Space<PageData> {
     });
   }
 
-  addSiblingBlock(
+  addSiblingBlocks(
     targetModel: BaseBlockModel,
-    blockProps: Partial<BaseBlockModel>,
+    props: Partial<BaseBlockModel> | Array<Partial<BaseBlockModel>>,
```
Comment (doodlewind) [inline packages/store/src/workspace/page.ts:497]: The type here should not be mixed IMO.

File: packages/store/src/workspace/page.ts [hunk 3]
```diff
@@ -313,6 +313,37 @@ export class Page extends Space<PageData> {
...
+    const ids: string[] = [];
+    blocks.forEach(block => {
+      const id = this.addBlockByFlavour<ALLProps, Flavour>(
+        block.flavour,
+        block.blockProps ?? {},
+        parent,
+        parentIndex
+      );
+      ids.push(id);
+      typeof parentIndex === 'number' && parentIndex++;
```
Comment (doodlewind) [inline packages/store/src/workspace/page.ts:341]: If no `parentIndex` is specified, will the subsequent blocks always being added to the first position?

Comment (doodlewind) [reply packages/store/src/workspace/page.ts:341]: If you are going to add new API in `Page`, I think the test case in `workspace.unit.spec.ts` should be provided altogether.

File: packages/blocks/src/components/blockhub.ts (hunk 1 above)
Comment (thorseraq) [reply packages/blocks/src/components/blockhub.ts:48]: It's used, can you explain?

File: packages/store/src/workspace/page.ts [hunk 4]
```diff
@@ -461,28 +492,44 @@ export class Page extends Space<PageData> {
...
   }
 
-  addSiblingBlock(
+  addSiblingBlocks(
     targetModel: BaseBlockModel,
-    blockProps: Partial<BaseBlockModel>,
+    props: Partial<BaseBlockModel> | Array<Partial<BaseBlockModel>>,
     direction: 'left' | 'right' = 'right'
-  ) {
+  ): string | string[] {
```
[the remark this answers is not in the document]
Comment (thorseraq) [reply packages/store/src/workspace/page.ts]: Updated at https://github.com/toeverything/blocksuite/pull/1188/commits/61967baa696e1136b59eea31dcf7bea499b3a54f

File: packages/store/src/workspace/page.ts (hunk 2 above)
Comment (thorseraq) [reply packages/store/src/workspace/page.ts]: updated at https://github.com/toeverything/blocksuite/pull/1188/commits/61967baa696e1136b59eea31dcf7bea499b3a54f

File: packages/store/src/workspace/page.ts (hunk 3 above)
Comment (thorseraq) [reply packages/store/src/workspace/page.ts:341]: >If you are going to add new API in Page, I think the test case in workspace.unit.spec.ts should be provided altogether.

working on it

Subsequent blocks will be added at the end page of top level (`const parentId = parent === null ? null : parent?.id ?? this.root?.id;`) in order (`const index = parentIndex ?? yChildren.length;`),

Comment (thorseraq) [reply packages/store/src/workspace/page.ts:341]: Updated at https://github.com/toeverything/blocksuite/pull/1188/commits/111120e449eed470bcb1b70809f491ec55d5a637

**Review by doodlewind — APPROVED** [review]

**Pull request merged**